Upgraded react-scripts to v2, removed npm vulnerabilities

slavabez opened the pull request:
Addressing issue #87
Note: Because I ran `npm install` before upgrading the versions, I had to delete node_modules and re-run npm install because the sass binary wouldn't run for me (Arch Linux, Latest Node 10).

react-scripts scripts were updated to v2, which has a number of improvements but some potential limitations, a short summary of what it means:

(+) Babel 7, better code-splitting with Webpack 4, Jest 23 with snapshot testing
(+) Out of the box support for Sass. Not sure if it's relevant since there is already commands to build sass, but the project can in theory be simplified to just import the sass files.
(+) Ability to target browsers using browserslist in package.json. Potentially creates smaller CSS files
(-) No support for Node 6 during development
(-) No support for IE 9-11 by default, extra setup required
I ran both npm run start and npm run start-fake and everything seemed to work fine.

If this is a deal-breaker I can try to install react-scripts@1.1.5, which would hopefully also fix the vulnerability warnings without upgrading to v2

Mcat12 commented on src/components/common/Sidebar.js:
```diff
@@ -35,6 +35,7 @@ export const navItem = (item, key, props) => (
 
 export const navDropdown = (item, key, props) => (
   <li key={key} className={dropDownClassList(item.url, props)}>
+    {/* eslint-disable-next-line */}
```
Could you fix the lint issue instead of disabling it? It might work if you set `href` to `#`.

slavabez replied:
Ofc, will do

slavabez commented:
Changed as requested. Removed the slint-disable-next-line, however it would not allow me to use `href="#"`, and suggested I use a button tag instead if I want that functionality without actually linking anywhere. Changed the a tag to a button, and added custom specific styles to keep it looking exactly like it did
